app/services/importers/ok_sos/split_files.rb

```diff
@@ -0,0 +1,121 @@
+module Importers
+  module OkSos
+    class SplitFiles < ApplicationService
+      attr_accessor :combined_zip_name
+
+      PREFIX_FILE_MAP = {
+        '01': 'entities',
+        '02': 'entity_addresses',
+        '03': 'agents',
+        '04': 'officers',
+        '05': 'names',
+        '06': 'associated_entities',
+        '07': 'stock_data',
+        '08': 'stock_infos',
+        '09': 'stock_types',
+        '10': 'filing_types',
+        '11': 'corp_statuses',
+        '12': 'corp_types',
+        '13': 'name_statuses',
+        '14': 'name_types',
+        '15': 'capacities',
+        '16': 'suffixes',
+        '17': 'corp_filings',
+        '18': 'audit_logs',
+        '99': 'etc_error'
+      }.stringify_keys
+
+      def initialize(combined_zip_name)
+        @combined_zip_name = combined_zip_name
+        @combined_file = nil
+        @csvs = {}
+      end
+
+      def perform
+        download_combined_csv
+        split_files
+        import_files
+      end
+
+      def import_files
+        [
+          :audit_logs,
+          :capacities,
+          :corp_statuses,
+          :corp_types,
+          :filing_types,
+          :entity_addresses,
+          :name_statuses,
+          :name_types,
+          :suffixes,
+          :stock_types,
+          :entities,
+          :corp_filings,
+          :names,
+          :stock_data,
+          :stock_infos,
+          :agents,
+          :associated_entities,
+          :officers
+        ].each do |file|
+          klass = Object.const_get("::Importers::OkSos::#{file.to_s.classify.pluralize}")
+          puts "importing #{klass}"
+          klass.perform(@csvs[file.to_s].path)
+        end
+      end
+
+      def split_files
+        columns = nil
```

Comment:
this is extra complex because it's reading the file row by row but some rows are broken up because of new line characters so need to be "merged"
I'm sure there's a better way to handle but I couldn't find...